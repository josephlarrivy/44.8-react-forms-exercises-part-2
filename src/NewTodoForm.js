import React, { useState } from 'react';

const TodoForm = ({addItem}) => {

    const INITIAL_STATE = 'new todo';

    const [formData, setFormData] = useState(INITIAL_STATE);

    const handleSubmit = evt => {
        evt.preventDefault();
        addItem({ ...formData });
        setFormData(INITIAL_STATE)
    };

    const handleChange = evt => {
        const {name, value} = evt.target;
        setFormData(data => ({
            ...data,
            [name]: value
        }));
    };


    return (
        <form onSubmit={handleSubmit}>
            <label htmlFor="item">Item: </label>
            <input
                required
                id="item"
                type="text"
                name="item"
                placeholder="Item"
                value={formData.item}
                onChange={handleChange}
            />
            <button>Submit</button>
        </form>
    )
}





export default TodoForm;