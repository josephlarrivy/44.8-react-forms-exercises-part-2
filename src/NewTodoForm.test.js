import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import NewTodoForm from './NewTodoForm'

test('renders the form', () => {
    render(<NewTodoForm />);
});

it("matches snapshot", function () {
    const { asFragment } = render(<NewTodoForm />);
    expect(asFragment()).toMatchSnapshot();
});
