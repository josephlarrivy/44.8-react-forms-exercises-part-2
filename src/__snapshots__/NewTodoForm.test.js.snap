// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <form>
    <label
      for="item"
    >
      Item: 
    </label>
    <input
      id="item"
      name="item"
      placeholder="Item"
      required=""
      type="text"
      value=""
    />
    <button>
      Submit
    </button>
  </form>
</DocumentFragment>
`;
